src/driver/mod.rs

```diff
@@ -0,0 +1,293 @@
+///
+/// - "NOT_FOUND\r\n" to indicate that the item with this key was not
+///   found.
+pub async fn delete<K>(
+    mut conn: PoolConnection<'_>,
+    key: K,
+    noreply: bool,
+) -> Result<Response, MemcacheError>
+where
+    K: AsRef<[u8]>,
+{
+    // <command name>
+    let _ = conn.write(COMMAND_DELETE).await?;
+    // <key>
+    let _ = conn.write_all(key.as_ref()).await?;
+
+    // [noreply]
+    if noreply {
+        // FYI: NO_REPLY_BYTES contains space before and new line after
+        let _ = conn.write(NO_REPLY_BYTES).await?;
+    } else {
+        let _ = conn.write(NEW_LINE_BYTES).await?;
+    }
+
+    // Flush command
+    let _ = conn.flush().await?;
+
+    // 64 bytes should be enough to address all storage responses
+    let mut buffer: BytesMut = BytesMut::with_capacity(64);
+
+    let _ = conn.read_buf(&mut buffer).await?;
+
+    match parser::parse_ascii_status(&buffer) {
+        Ok((_left, result)) => Ok(result),
+        Err(e) => Err(MemcacheError::Nom(format!("{}", e))),
+    }
+}
+
+/// touch <key> <exptime> [noreply]\r\n
+///
+///
+/// The response line to this command can be one of:
+///
+/// - "TOUCHED\r\n" to indicate success
+///
+/// - "NOT_FOUND\r\n" to indicate that the item with this key was not
+///   found.
+pub async fn touch<K>(
+    mut conn: PoolConnection<'_>,
+    key: K,
+    expiration: impl Into<Option<Duration>>,
+    noreply: bool,
+) -> Result<Response, MemcacheError>
+where
+    K: AsRef<[u8]>,
+{
+    // <command name>
+    let _ = conn.write(COMMAND_TOUCH).await?;
+    // <key>
+    let _ = conn.write_all(key.as_ref()).await?;
+    let _ = conn.write(EMPTY_SPACE_BYTES).await?;
+
+    // <exptime>
+    let exptime = expiration.into().map(|d| d.as_secs()).unwrap_or(0);
+    let _ = conn.write(exptime.to_string().as_ref()).await?;
+    let _ = conn.write(EMPTY_SPACE_BYTES).await?;
+
+    // [noreply]
+    if noreply {
+        // FYI: NO_REPLY_BYTES contains space before and new line after
+        let _ = conn.write(NO_REPLY_BYTES).await?;
+    } else {
+        let _ = conn.write(NEW_LINE_BYTES).await?;
+    }
+
+    // Flush command
+    let _ = conn.flush().await?;
+
+    // 64 bytes should be enough to address all storage responses
+    let mut buffer: BytesMut = BytesMut::with_capacity(64);
+
+    let _ = conn.read_buf(&mut buffer).await?;
+
+    match parser::parse_ascii_status(&buffer) {
+        Ok((_left, result)) => Ok(result),
+        Err(e) => Err(MemcacheError::Nom(format!("{}", e))),
+    }
+}
+
+/// version\r\n
+///
+///
+/// "VERSION <version>\r\n", where <version> is the version string for the
+pub async fn version(conn: &mut PoolConnection<'_>) -> Result<String, MemcacheError> {
+    // <command name>
+    let _ = conn.write(COMMAND_VERSION).await?;
+
+    // Flush command
+    let _ = conn.flush().await?;
+
+    // 64 bytes should be enough to address all storage responses
+    let mut buffer: BytesMut = BytesMut::with_capacity(64);
```

Comment:
I see 64 bytes being reused in a couple of places, consider declaring a const up at the file?